### PR TITLE
[API] Add Neural Network Model Compile C API

### PR DESCRIPTION
**Pull request by jijoongmoon**
Add neural network model compile c api
  : ml_nnmodel_compile (ml_nnmodel_h model)
  . check validation of model
  . if it is Ok, then run NeuralNetwork.ini()

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: api/capi/src/nntrainer.cpp [hunk 1]
```diff
@@ -99,6 +99,20 @@ int ml_nnmodel_construct_with_conf(const char *model_conf,
...
 
+int ml_nnmodel_compile(ml_nnmodel_h model) {
+  int status = ML_ERROR_NONE;
+  ml_nnmodel *nnmodel;
+
+  ML_NNTRAINER_CHECK_MODEL_VALIDATION(nnmodel, model);
+  nntrainer::NeuralNetwork *NN;
+  NN = nnmodel->network;
+  status = NN->checkValidation();
+  if (!status)
```
Comment (zhoonit) [inline api/capi/src/nntrainer.cpp:110]: NO big deal though. IMO, `status != ML_ERROR_NONE` is more clear.

File: nntrainer/include/layers.h [hunk 2]
```diff
@@ -47,7 +47,7 @@ typedef enum { COST_MSR, COST_ENTROPY, COST_UNKNOWN } CostType;
  *            3. Unknown
  */
 typedef enum {
-  ACT_TANH,
+  ACT_TANH = 1,
```
Comment (zhoonit) [inline nntrainer/include/layers.h:50]: `tanh` is 0 in the documentaion.

File: nntrainer/src/layers.cpp [hunk 3]
```diff
@@ -129,6 +129,24 @@ int Layer::setOptimizer(Optimizer opt) {
...
+    ml_loge("Error: Layer type is unknown");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!activation_type || activation_type == ACT_UNKNOWN) {
+    ml_loge("Error: Have to set activation for this layer");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+  if (!batch || !width || !height) {
+    ml_loge("Error: Tensor Dimension must be set before initalization");
```
Comment (zhoonit) [inline nntrainer/src/layers.cpp:144]:
```suggestion
    ml_loge("Error: Tensor Dimension must be set before initialization");
```

File: nntrainer/src/layers.cpp [hunk 4]
```diff
@@ -129,6 +129,24 @@ int Layer::setOptimizer(Optimizer opt) {
...
+  if (!type || type == LAYER_UNKNOWN) {
+    ml_loge("Error: Layer type is unknown");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (!activation_type || activation_type == ACT_UNKNOWN) {
+    ml_loge("Error: Have to set activation for this layer");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+  if (!batch || !width || !height) {
```
Comment (zhoonit) [inline nntrainer/src/layers.cpp:143]: I think it is safer to validate `under 0` even though it is already checked at init phase.
```suggestion
  if (batch <= 0 || width <= 0 || height <= 0) {
```

Comment (jijoongmoon) [reply nntrainer/src/layers.cpp]: It's unsigned int type. I will change ```batch == 0```

File: nntrainer/src/layers.cpp (hunk 3 above)
Comment (jijoongmoon) [reply nntrainer/src/layers.cpp]: Thanks. Fixed.

File: nntrainer/include/layers.h (hunk 2 above)
Comment (jijoongmoon) [reply nntrainer/include/layers.h:50]: fixed.

File: api/capi/src/nntrainer.cpp (hunk 1 above)
Comment (jijoongmoon) [reply api/capi/src/nntrainer.cpp]: Fixed. Thanks.

**Pull request merged**